ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh

```diff
@@ -153,16 +140,46 @@ ibm_cloud_cis_crn = "${IBMCLOUD_CIS_CRN}"
 ibm_cloud_tgw              = "${workspace_name}-tg"
 EOF
 
-  cp "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-mac-upi.tfvars "${SHARED_DIR}"/var-mac-upi.tfvars
-  cat "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-mac-upi.tfvars
+  cp "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-multi-arch-upi.tfvars "${SHARED_DIR}"/var-multi-arch-upi.tfvars
+  echo "UPI TFVARS created: ${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-multi-arch-upi.tfvars
 }
 
 function create_upi_powervs_cluster() {
   cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ || true
   # Dev Note: https://github.com/ocp-power-automation/openshift-install-power/blob/devel/openshift-install-powervs#L767C1-L767C145
   # May trigger the redaction
   OUTPUT="yes"
-  ./openshift-install-powervs create -var-file var-mac-upi.tfvars -ignore-os-checks -verbose | sed '/.*client-certificate-data*/d; /.*token*/d; /.*client-key-data*/d; /- name: /d; /Login to the console with user/d' | \
+  # short-circuit to download and install terraform
+  echo "Attempting to install terraform using gzip"
+  curl -L -o "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.gz -L https://releases.hashicorp.com/terraform/"${TERRAFORM_VERSION}"/terraform_"${TERRAFORM_VERSION}"_linux_amd64.zip \
+    && gunzip "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.gz \
+    && chmod +x "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform \
+    || true
+
+  if [ ! -f "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform ]
+  then
+    echo "Manually building the terraform code"
+    # upgrade go to GO_VERSION
+    if [ -z "$(command -v go)" ]
+    then
+      echo "go is not installed, proceed to installing go"
+      cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ && wget -q https://go.dev/dl/go"${GO_VERSION}".linux-amd64.tar.gz && tar -C /tmp -xzf go"${GO_VERSION}".linux-amd64.tar.gz \
+        && export PATH=$PATH:/tmp/go/bin && export GOCACHE=/tmp/go && export GOPATH=/tmp/go && export GOMODCACHE="/tmp/go/pkg/mod"
+      if [ -z "$(command -v go)" ]
+      then
+        echo "Installed go successfully"
+      fi
+    fi
+
+    # build terraform from source using TERRAFORM_VERSION
+    echo "terraform is not installed, proceed to installing terraform"
+    cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ && curl -L https://github.com/hashicorp/terraform/archive/refs/tags/v"${TERRAFORM_VERSION}".tar.gz -o "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.tar.gz \
+      && tar -xzf "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.tar.gz && cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform-"${TERRAFORM_VERSION}" \
+      && go build -ldflags "-w -s -X 'github.com/hashicorp/terraform/version.dev=no'" -o bin/ . && cp bin/terraform "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform
+  fi
+  cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ || true
+  terraform apply -auto-approve -var-file var-multi-arch-upi.tfvars | \
```

Comment:
```suggestion
  terraform apply -auto-approve -var-file var-multi-arch-upi.tfvars | \
```
is terraform in the path?